Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java

```diff
@@ -486,9 +486,14 @@ public static BranchUniversalObject getReferredBranchUniversalObject() {
         Branch branchInstance = Branch.getInstance();
         try {
             if (branchInstance != null && branchInstance.getLatestReferringParams() != null) {
+                // Check if link clicked. Unless deepvlink debug enabled return null if there is no link click
                 if (branchInstance.getLatestReferringParams().has("+clicked_branch_link") && branchInstance.getLatestReferringParams().getBoolean("+clicked_branch_link")) {
                     branchUniversalObject = createInstance(branchInstance.getLatestReferringParams());
                 }
+                // If debug params are set then send BUO object even if link click is false
+                else if (branchInstance.getDeeplinkDebugParams() != null && branchInstance.getDeeplinkDebugParams().length() > 0) {
```

Comment:
I think there should be a flag to toggle this mode, since I'm worried about the case where the above line is false `(+clicked_branch_link == 0)` and someone ships an .apk without commenting out: 

`Branch.getInstance().setDeepLinkDebugMode(JSONObject ...)`

Thoughts?